lib/aur-build--sync

```diff
@@ -0,0 +1,62 @@
+#!/bin/perl -T
+# This script can be used in NOPASSWD rules (sudoers) or similar (doas/setuid
+# wrappers) to upgrade packages in a local repository without password prompt.
+# Use basic measures from perlsec(1perl) to limit security impact.
+use strict;
+use warnings;
+use v5.20;
+my $argv0 = 'build--sync';
+
+# minimal path
+$ENV{PATH} = "/bin:/usr/bin";
```

Comment:
It may not be necessary here but there is a `getconf(3)` attribute called `PATH` or `_CS_PATH` (as used by `command -p`) which is supposed to represent the default system `PATH` where all POSIX utilties can be found.

Reply:
> It may not be necessary here but there is a getconf(3) attribute called PATH or _CS_PATH (as used by command -p) which is supposed to represent the default system PATH where all POSIX utilties can be found.

`perl` has a `POSIX` module, so it might be queried this way.

> However consider that many of these security practices are something the sudo/doas tool should be responsible for and so if a user configures those to allow certain things it might not be nice to undermine them later; unintuitively.

Good point. I could introduce a `--reset-env` option or (conversely) `--preserve-env` to address this.

I initially came to `perl -T` after seeing older versions of `opendoas` [would not reset PATH](https://github.com/advisories/GHSA-59xc-99wg-3hgf). `perl -T` forces you to think on which data is trusted and which isn't, and for that alone I find it useful.

Reply:
In the end I went with a simple `bash` script and left things to the user.